### PR TITLE
javascript: split ParseStatement into multiple functions

### PR DESCRIPTION
**Pull request by jafl**
also improve code re-use

### CODE REVIEW

**Comment by jafl** [comment]
@masatake I can't figure out why the MacOS tests are failing.  Homebrew claims to have python3.9 when I use `brew search`

**Comment by masatake** [comment]
I'm inspecting the issue now.

**Comment by masatake** [comment]
See  #3466.
Could you rebase on the latest master?

**Comment by jafl** [comment]
@masatake Only 2 checks?

File: parsers/jscript.c [hunk 1]
```diff
@@ -1360,7 +1360,7 @@ static int parseMethodsInAnonymousObject (tokenInfo *const token)
 
 	tokenInfo *const anon_object = newToken ();
 	copyToken (anon_object, token, true);
-	anonGenerate (anon_object->string, "anonymousObject", JSTAG_VARIABLE);
+	anonGenerate (anon_object->string, "anonymousObject_", JSTAG_VARIABLE);
```
Comment (masatake) [inline parsers/jscript.c:1363]: Do we need this in this commit?

This commit is marked as "refactor". This change is not suitable for such a commit.
I expect if a commit is marked "refactor", most of all test cases are passed successfully without changing the test cases. (An exception is adding a new test case.)

File: parsers/jscript.c [hunk 2]
```diff
@@ -1755,47 +1756,58 @@ static void parseFunction (tokenInfo *const token)
 		is_generator = true;
 		readToken (name);
 	}
+
 	if (isType (name, TOKEN_OPEN_PAREN))
 	{
 		/* anonymous function */
 		copyToken (token, name, false);
-		anonGenerate (name->string, "anonymousFunction", JSTAG_FUNCTION);
+		anonGenerate (name->string, "anonymousFunction_", JSTAG_FUNCTION);
```
Comment (masatake) [inline parsers/jscript.c:1764]: Do we need this in this commit?

If you think we need this one, let's make a commit for introducing "_".
You can prepend such a commit to this pull request.
In such a case, I would like you to write the background or reason why "_" is needed in its commit log. I don't say introducing "_" breaks compatibilities. However, I don't say introducing "_" is a small change. Unlike "refactor", it is user-visible change. Therefore, I would like you to explain this to users.

File: Units/review-needed.r/simple.js.t/input.js [hunk 3]
```diff
@@ -40,8 +39,6 @@ function validFunctionThree(a,b) {
 
 var my_global_var1 = 33;
 
-function extra.validFunctionFour(a,b) {}
```
Comment (masatake) [inline Units/review-needed.r/simple.js.t/input.js:43]: Why do we have to remove this?

A. the limitation of this refactor. we will support this syntax in the future.
For A, you can move this fragment to ".b" test case.

B. the parser should not support this syntax for some reason.
For B, could you write the reason or background to the commit log?
This is a "refactor" commit. For such a commit, we expect all test cases for the parser should be passed without modification.

File: parsers/jscript.c [hunk 4]
```diff
@@ -1755,47 +1756,58 @@ static void parseFunction (tokenInfo *const token)
 		is_generator = true;
 		readToken (name);
 	}
+
```
Comment (masatake) [inline parsers/jscript.c:1759]: Minor comment. This addition is not a must.

File: parsers/jscript.c [hunk 5]
```diff
@@ -1755,47 +1756,58 @@ static void parseFunction (tokenInfo *const token)
...
-	findCmdTerm (token, false, false);
+	if ( lhs_name == NULL )
+		findCmdTerm (token, false, false);
 
  cleanUp:
 	vStringDelete (signature);
 	deleteToken (name);
 
 	TRACE_LEAVE();
+
```
Comment (masatake) [inline parsers/jscript.c:1809]: Minor comment. This addition is not a must.

File: parsers/jscript.c [hunk 6]
```diff
@@ -2255,6 +2267,7 @@ static bool parseES6Class (tokenInfo *const token, const tokenInfo *targetName)
 	deleteToken (className);
 
 	TRACE_LEAVE();
+
```
Comment (masatake) [inline parsers/jscript.c:2270]: Minor comment. This addition is not a must.

**Review by masatake — CHANGES_REQUESTED** [review]
Great work.

Do you know git add -p?
If you want to insert empty lines to adjust the code to your favorite, you can make a separate commit with 'git add -p'.

File: Units/review-needed.r/simple.js.t/input.js (hunk 3 above)
Comment (jafl) [reply Units/review-needed.r/simple.js.t/input.js:43]: This is invalid JavaScript.  I will create a separate commit for it.

**Review by masatake — APPROVED** [review]

**Pull request merged**